### PR TITLE
docs(sdk): add loading and error states to docs

### LINKED ISSUE #49194: Document loader and error components in sdk's README
We should add the loader and error components to the README.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/49194

Documents how to customize the loading and error states in the sdk

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 1]
```diff
@@ -1224,6 +1224,19 @@ async function fetchRequestToken(url) {
...
+
+```tsx
+import { defineEmbeddingSdkConfig } from "@metabase/embedding-sdk-react";
+
+const config = defineEmbeddingSdkConfig({
+  loaderComponent: () => <div>Analytics is loading...</div>,
+  errorComponent: ({ message }) => <div>There was an error: {message}</div>,
+});
+```
+
```
Comment (jeff-bruemmer) [inline enterprise/frontend/src/embedding-sdk/README.md:1239]:
```suggestion
```

I'm going to be changing this README to point to the docs soon.

```

File: enterprise/frontend/src/embedding-sdk/README.md [hunk 2]
```diff
@@ -1224,6 +1224,19 @@ async function fetchRequestToken(url) {
 const config = defineEmbeddingSdkConfig({ fetchRequestToken });
 ```
 
+### Customizing loader and error components
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/README.md:1227]: Note that we should have a PR soon that removes most of the content from README.md, and point to https://www.metabase.com/docs/latest/embedding/sdk/quickstart instead as the authoritative source of SDK documentation.

**Review by jeff-bruemmer — APPROVED** [review]
Thanks, @heypoom!

File: enterprise/frontend/src/embedding-sdk/README.md (hunk 1 above)
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/README.md:1239]: @jeff-bruemmer Thanks! I'll avoid adding more things to `README.md` in the meantime and just add them to `docs/embedding` then 🙏🏻

**Pull request merged**